.travis.yml

```diff
@@ -1,53 +1,94 @@
+    - env: ENV=vim74-trusty
+      dist: trusty
       rvm: 1.8.7
       addons: { apt: { packages: [vim-nox] } }
-    - env: ENV=python
+    - env: ENV=vim74-xenial
+      dist: xenial
+      rvm: 2.3.0
+      addons: { apt: { packages: [vim-nox] } }
+    - env: ENV=vim80-bionic
+      dist: bionic
+      rvm: 2.5.1
+      addons: { apt: { packages: [vim-nox] } }
+    - env: ENV=python-trusty
+      dist: trusty
       rvm: 1.8.7
       addons: { apt: { packages: [python2.7-dev] } }
-    - env: ENV=python3
+    - env: ENV=python-xenial
+      dist: xenial
+      rvm: 2.3.0
+      addons: { apt: { packages: [python2.7-dev] } }
+    - env: ENV=python-bionic
+      dist: bionic
+      rvm: 2.5.1
+      addons: { apt: { packages: [python2.7-dev] } }
+    - env: ENV=python3-trusty
+      dist: trusty
       rvm: 1.8.7
       addons: { apt: { packages: [python3-dev] } }
-    - env: ENV=ruby18
+    - env: ENV=python3-xenial
+      dist: xenial
+      rvm: 2.3.0
+      addons: { apt: { packages: [python3-dev] } }
+    - env: ENV=python3-bionic
+      dist: xenial
+      rvm: 2.5.1
+      addons: { apt: { packages: [python3-dev] } }
+    - env: ENV=ruby-trusty
+      dist: trusty
       rvm: 1.8.7
-    - env: ENV=ruby20
-      rvm: 2.0.0
-    - env: ENV=neovim
+    - env: ENV=ruby-xenial
+      dist: xenial
+      rvm: 2.3.0
+    - env: ENV=ruby-bionic
+      dist: xenial
+      rvm: 2.5.1
+    - env: ENV=neovim-trusty
+      dist: trusty
+      addons: {apt: {packages: [neovim], sources: [{sourceline: 'ppa:neovim-ppa/unstable'}]}}
+    - env: ENV=neovim-xenial
+      dist: xenial
+      addons: {apt: {packages: [neovim], sources: [{sourceline: 'ppa:neovim-ppa/unstable'}]}}
+    - env: ENV=neovim-bionic
+      dist: bionic
+      addons: {apt: {packages: [neovim], sources: [{sourceline: 'ppa:neovim-ppa/unstable'}]}}
     - env: ENV=vim8
+      dist: trusty
+    - env: ENV=vim8
+      dist: xenial
+    - env: ENV=vim8
+      dist: bionic
 install: |
   git config --global user.email "you@example.com"
   git config --global user.name "Your Name"
 
-  if [ "$ENV" == "vim74" ]; then
-    mkdir -p ${DEPS}/bin
-    ln -s /usr/bin/vim.nox ${DEPS}/bin/vim
-    return
-  elif [ "$ENV" == "neovim" ]; then
-    # https://github.com/neovim/bot-ci#nightly-builds
-    eval "$(curl -Ss https://raw.githubusercontent.com/neovim/bot-ci/master/scripts/travis-setup.sh) nightly-x64"
-    mkdir -p ${DEPS}/bin
-    ln -s $(which nvim) ${DEPS}/bin/vim
-    export VADER_OUTPUT_FILE=/dev/stderr
-    return
-  fi
-
   C_OPTS="--prefix=$DEPS --with-features=huge --disable-gui "
   case "$ENV" in
-    python)
+    neovim-*)
+      mkdir -p ${DEPS}/bin
+      ln -s /usr/bin/nvim ${DEPS}/bin/vim
+      export VADER_OUTPUT_FILE=/dev/stderr
+      return
+      ;;
+    vim74-* | vim80-*)
+      mkdir -p ${DEPS}/bin
+      ln -s /usr/bin/vim.nox ${DEPS}/bin/vim
+      return
+      ;;
+    python-*)
       C_OPTS+=--enable-pythoninterp
       ;;
-    python3)
+    python3-*)
```

Comment:
Is this suppose to test on Vim nightly (8.1) or the Vim version that Ubuntu has (7.4, 8.0)?

Reply:
The latter. Ruby/Python installers are no longer used on the latest versions of Vim.